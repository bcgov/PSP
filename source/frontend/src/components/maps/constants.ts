import { LatLngBounds } from 'leaflet';

export const DEFAULT_MAP_ZOOM = 6;
/** rough center of bc Itcha Ilgachuz Provincial Park */
export const defaultLatLng = {
  lat: 52.81604319154934,
  lng: -124.67285156250001,
};

// default BC map bounds
export const defaultBounds = new LatLngBounds(
  [60.09114547, -119.49609429],
  [48.78370426, -139.35937554],
);

export const ONE_HUNDRED_METER_PRECISION = 0.0001;
