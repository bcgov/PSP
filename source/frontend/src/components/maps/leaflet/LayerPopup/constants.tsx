import { Feature } from 'geojson';

import { exists, pidFormatter } from '@/utils';

import { PopupContentConfig } from './components/LayerPopupContent';

export const parcelLayerPopupConfig: PopupContentConfig = {
  PID: { label: 'Parcel PID', display: (data: { [key: string]: any }) => pidFormatter(data?.PID) },
  PIN: { label: 'Parcel PIN', display: (data: { [key: string]: any }) => data.PIN },
  PLAN_NUMBER: {
    label: 'Plan number',
    display: (data: { [key: string]: any }) => data.PLAN_NUMBER,
  },
  OWNER_TYPE: { label: 'Owner type', display: (data: { [key: string]: any }) => data.OWNER_TYPE },
  MUNICIPALITY: {
    label: 'Municipality',
    display: (data: { [key: string]: any }) => data.MUNICIPALITY,
  },
  FEATURE_AREA_SQM: {
    label: 'Area',
    display: (data: { [key: string]: any }) => (
      <>
        {data.FEATURE_AREA_SQM} m<sup>2</sup>
      </>
    ),
  },
};

// Utility function that dynamically creates a PopupContentConfig for a feature.
// Transforms "THE_PARAM" to "The param"
export const getDynamicFeatureConfig: (feature: Feature) => PopupContentConfig = (
  feature: Feature,
) => {
  // For debug enable this
  /*
  config['Feature Type'] = {
    label: 'Feature Type',
    display: (data: { [key: string]: any }) => feature.id,
  };
  */
  const config: PopupContentConfig = {};
  Object.keys(feature.properties).forEach(key => {
    config[key] = {
      label: key.charAt(0).toUpperCase() + key.slice(1).toLowerCase().replaceAll('_', ' '),
      display: (data: { [key: string]: any }) => data[key],
    };
  });
  // This is not being mapped in the typed feature data
  if (exists(config['SE_ANNO_CAD_DATA'])) {
    delete config['SE_ANNO_CAD_DATA'];
  }
  return config;
};

export const municipalityLayerPopupConfig: PopupContentConfig = {
  ADMIN_AREA_GROUP_NAME: {
    label: 'Administration Area',
    display: (data: { [key: string]: any }) =>
      `${data.ADMIN_AREA_GROUP_NAME} (${data.CHANGE_REQUESTED_ORG})`,
  },
};

export const highwayLayerPopupConfig: PopupContentConfig = {
  PROVINCIAL_PUBLIC_HIGHWAY_ID: {
    label: 'Id',
    display: (data: { [key: string]: any }) => data.PROVINCIAL_PUBLIC_HIGHWAY_ID,
  },
  GLOBALID: { label: 'Global Id', display: (data: { [key: string]: any }) => data.GLOBALID },
  UNIQUE_ID: { label: 'Unique Id', display: (data: { [key: string]: any }) => data.UNIQUE_ID },
  HYPERLINK: {
    label: 'Hyperlink to plan documents',
    display: (data: { [key: string]: any }) => (
      <a href={`${data.HYPERLINK}`} target="_blank" rel="noreferrer">
        {decodeURI(data.HYPERLINK)}
      </a>
    ),
  },
  PLAN_ANNOTATION: {
    label: 'Plan annotation',
    display: (data: { [key: string]: any }) => data.PLAN_ANNOTATION,
  },
  MOTI_PLAN: { label: 'MOTI plan', display: (data: { [key: string]: any }) => data.MOTI_PLAN },
  VETTING_STATUS: {
    label: 'Vetting Status',
    display: (data: { [key: string]: any }) => data.VETTING_STATUS,
  },
  SHAPE_TYPE: { label: 'Shape type', display: (data: { [key: string]: any }) => data.SHAPE_TYPE },
  PENDING_CLASSIFICATION: {
    label: 'Pending classification',
    display: (data: { [key: string]: any }) => data.PENDING_CLASSIFICATION,
  },
  MOTI_FILE: { label: 'MOTI file', display: (data: { [key: string]: any }) => data.MOTI_FILE },
  GAZETTE_PUBLISHED_DATE: {
    label: 'Gazette published date',
    display: (data: { [key: string]: any }) => data.GAZETTE_PUBLISHED_DATE,
  },
  GAZETTE_PUBLISHED_ON: {
    label: 'Gazette published on',
    display: (data: { [key: string]: any }) => data.GAZETTE_PUBLISHED_ON,
  },
  PROV_PUBLIC_HIGHWAY_TYPE: {
    label: 'Prov public highway type',
    display: (data: { [key: string]: any }) => data.PROV_PUBLIC_HIGHWAY_TYPE,
  },
  ORDER_IN_COUNCIL: {
    label: 'Order in council',
    display: (data: { [key: string]: any }) => data.ORDER_IN_COUNCIL,
  },
  SHORT_LEGAL_DESCRIPTION: {
    label: 'Short legal description',
    display: (data: { [key: string]: any }) => data.SHORT_LEGAL_DESCRIPTION,
  },
  SURVEY_PLAN: {
    label: 'Survey plan',
    display: (data: { [key: string]: any }) => data.SURVEY_PLAN,
  },
  PARENT_PARCEL: {
    label: 'Parent parcel',
    display: (data: { [key: string]: any }) => data.PARENT_PARCEL,
  },
  PARCELMAPBC_COMMENT: {
    label: 'Parcelmapbc comment',
    display: (data: { [key: string]: any }) => data.PARCELMAPBC_COMMENT,
  },
  CREATE_TIME: {
    label: 'Create time',
    display: (data: { [key: string]: any }) => data.CREATE_TIME,
  },
  UPDATE_TIME: {
    label: 'Update time',
    display: (data: { [key: string]: any }) => data.UPDATE_TIME,
  },
  CREATE_USER: {
    label: 'Create user',
    display: (data: { [key: string]: any }) => data.CREATE_USER,
  },
  UPDATE_USER: {
    label: 'Update user',
    display: (data: { [key: string]: any }) => data.UPDATE_USER,
  },
  LTSA_CREATE_TIME: {
    label: 'Ltsa create time',
    display: (data: { [key: string]: any }) => data.LTSA_CREATE_TIME,
  },
  LTSA_UPDATE_TIME: {
    label: 'Ltsa update time',
    display: (data: { [key: string]: any }) => data.LTSA_UPDATE_TIME,
  },
  LTSA_CREATE_USER: {
    label: 'Ltsa create user',
    display: (data: { [key: string]: any }) => data.LTSA_CREATE_USER,
  },
  LTSA_UPDATE_USER: {
    label: 'Ltsa update user',
    display: (data: { [key: string]: any }) => data.LTSA_UPDATE_USER,
  },
  RELATED_GAZETTES: {
    label: 'Related gazettes',
    display: (data: { [key: string]: any }) => data.RELATED_GAZETTES,
  },
  RELATIVE_PATH: {
    label: 'Relative path',
    display: (data: { [key: string]: any }) => data.RELATIVE_PATH,
  },
};
