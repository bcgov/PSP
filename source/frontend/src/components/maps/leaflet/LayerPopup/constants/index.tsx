import { pidFormatter } from 'utils';

export const MUNICIPALITY_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP';
export const PARCELS_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_FA_SVW/wfs?service=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_FA_SVW';
export const ELECTORAL_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_ADMIN_BOUNDARIES.EBC_ELECTORAL_DISTS_BS10_SVW/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_ADMIN_BOUNDARIES.EBC_ELECTORAL_DISTS_BS10_SVW';
export const REGIONAL_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_REGIONAL_DISTRICTS_SP/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_REGIONAL_DISTRICTS_SP';

export const MOTI_REGION_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_ADMIN_BOUNDARIES.TADM_MOT_REGIONAL_BNDRY_POLY/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_ADMIN_BOUNDARIES.TADM_MOT_REGIONAL_BNDRY_POLY';
export const HWY_DISTRICT_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_ADMIN_BOUNDARIES.TADM_MOT_DISTRICT_BNDRY_POLY/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_ADMIN_BOUNDARIES.TADM_MOT_DISTRICT_BNDRY_POLY';
export const ALR_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_LEGAL_ADMIN_BOUNDARIES.OATS_ALR_BOUNDARY_LINES_SVW/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_LEGAL_ADMIN_BOUNDARIES.OATS_ALR_BOUNDARY_LINES_SVW';
export const INDIAN_RESERVES_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_ADMIN_BOUNDARIES.ADM_INDIAN_RESERVES_BANDS_SP/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_ADMIN_BOUNDARIES.ADM_INDIAN_RESERVES_BANDS_SP';
export const PIMS_BOUNDARY_LAYER_URL =
  '/ogs-internal/ows?service=wfs&request=GetFeature&typeName=PIMS_PROPERTY_BOUNDARY_VW&outputformat=json&version=2.0.0';

export const parcelLayerPopupConfig = {
  PID: { label: 'Parcel PID:', display: (data: any) => pidFormatter(data?.PID) },
  PIN: { label: 'Parcel PIN:', display: (data: any) => data.PIN },
  PLAN_NUMBER: { label: 'Plan number:', display: (data: any) => data.PLAN_NUMBER },
  OWNER_TYPE: { label: 'Owner type:', display: (data: any) => data.OWNER_TYPE },
  MUNICIPALITY: { label: 'Municipality:', display: (data: any) => data.MUNICIPALITY },
  FEATURE_AREA_SQM: {
    label: 'Area:',
    display: (data: any) => (
      <>
        {data.FEATURE_AREA_SQM} m<sup>2</sup>
      </>
    ),
  },
};

export const municipalityLayerPopupConfig = {
  ADMIN_AREA_GROUP_NAME: {
    label: 'Administration Area:',
    display: (data: any) => `${data.ADMIN_AREA_GROUP_NAME} (${data.CHANGE_REQUESTED_ORG})`,
  },
};
