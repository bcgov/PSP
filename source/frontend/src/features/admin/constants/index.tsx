import { ColumnWithProps } from 'components/Table';
import { IOrganization, IOrganizationRecord } from 'interfaces';
import * as React from 'react';
import { Link } from 'react-router-dom';
import { CellProps } from 'react-table';

export const columnDefinitions: ColumnWithProps<IOrganizationRecord>[] = [
  {
    Header: 'Organization name',
    accessor: 'name',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Short name',
    accessor: 'code',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Description',
    accessor: 'description',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Parent Organization',
    accessor: 'parent',
    align: 'left',
    clickable: true,
    Cell: (props: CellProps<IOrganization>) => {
      return (
        <Link to={`/admin/organization/${props.row.original.parentId}`}>
          {props.row.original.parent}
        </Link>
      );
    },
  },
];
