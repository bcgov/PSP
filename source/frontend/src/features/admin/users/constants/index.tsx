import { ReactComponent as Active } from 'assets/images/active.svg';
import { ReactComponent as Inactive } from 'assets/images/inactive.svg';
import { ColumnWithProps } from 'components/Table';
import { DateTimeCell } from 'components/Table/DateCell';
import { Link } from 'react-router-dom';
import { CellProps } from 'react-table';
import { stringToFragment } from 'utils';

import { RowActions } from '../components/RowActions';
import { FormUser } from '../models';

export const getUserColumns = (refresh: () => void): ColumnWithProps<FormUser>[] => [
  {
    Header: 'Active',
    accessor: 'isDisabled',
    align: 'left',
    clickable: true,
    sortable: true,
    width: 60,
    Cell: (props: CellProps<FormUser>) =>
      props.row.original.isDisabled ? <Inactive /> : <Active />,
  },
  {
    Header: 'IDIR/BCeID',
    accessor: 'businessIdentifierValue',
    align: 'left',
    minWidth: 150,
    Cell: (props: CellProps<FormUser>) => {
      return (
        <Link to={`/admin/user/${props.row.original.id}`}>
          {props.row.original.businessIdentifierValue}
        </Link>
      );
    },
    sortable: true,
  },
  {
    Header: 'First name',
    accessor: 'firstName',
    align: 'left',
    clickable: true,
    sortable: true,
  },
  {
    Header: 'Last name',
    accessor: 'surname',
    align: 'left',
    clickable: true,
    sortable: true,
  },
  {
    Header: 'Email',
    accessor: 'email',
    align: 'left',
    clickable: true,
    sortable: true,
    minWidth: 200,
  },
  {
    Header: 'Position',
    accessor: 'position',
    align: 'left',
    clickable: true,
    sortable: true,
  },

  {
    Header: 'Roles',
    accessor: 'roles',
    align: 'left',
    clickable: true,
    minWidth: 200,
    Cell: (props: CellProps<FormUser>) => {
      const rolesString = props.row?.original?.roles?.map(userRole => userRole?.name);
      return stringToFragment(rolesString?.join(', '));
    },
  },
  {
    Header: 'MoTI region(s)',
    accessor: 'regions',
    align: 'left',
    clickable: true,
    minWidth: 200,
    Cell: (props: CellProps<FormUser>) => {
      const regionsString = props.row?.original?.regions?.map(
        regionUser => regionUser?.description,
      );
      return stringToFragment(regionsString?.join(', '));
    },
  },
  {
    Header: 'Last login',
    accessor: 'lastLogin',
    align: 'left',
    clickable: true,
    minWidth: 100,
    Cell: DateTimeCell,
  },
  {
    Header: ' ',
    Cell: (props: CellProps<FormUser>) => {
      return <RowActions {...props} refresh={refresh} />;
    },
    width: 75,
  },
];
