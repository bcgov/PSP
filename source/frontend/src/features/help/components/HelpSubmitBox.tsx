import './HelpSubmitBox.scss';

import * as React from 'react';

import HelpRequestForm from '../forms/HelpRequestForm';

interface IHelpSubmitBoxProps {
  /** the keycloak user display name */
  user: string;
  /** the keycloak user email */
  email: string;
  /** the name of the current page, used to provide context in the email generated by the ticket form. ie. Landing Page. */
  page: string;
  /** set the mailto of the parent based on the ticket form content. */
  setMailto: Function;
}

/**
 * A component that displays a list of Help Ticket types as well as a form corresponding to the active Help Ticket type
 */
const HelpSubmitBox: React.FunctionComponent<React.PropsWithChildren<IHelpSubmitBoxProps>> = ({
  setMailto,
  ...rest
}) => {
  return <HelpRequestForm formValues={rest} setMailto={setMailto}></HelpRequestForm>;
};

export default HelpSubmitBox;
