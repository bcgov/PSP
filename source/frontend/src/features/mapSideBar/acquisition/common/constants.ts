export const cannotEditMessage =
  'This section of the file is in a non-editable state. Contact a System administrator to change it to an editable state';
