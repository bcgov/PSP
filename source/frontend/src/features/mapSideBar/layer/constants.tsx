import { Feature } from 'geojson';
import { FaExternalLinkAlt } from 'react-icons/fa';

import { ApiGen_Concepts_Address } from '@/models/api/generated/ApiGen_Concepts_Address';
import { exists, formatApiAddress, getAreaUnit, pidFormatter } from '@/utils';

import { ContentConfig } from './LayerContent';

export const parcelLayerConfig: ContentConfig = {
  PID: { label: 'Parcel PID', display: (data: { [key: string]: any }) => pidFormatter(data?.PID) },
  PIN: { label: 'Parcel PIN', display: (data: { [key: string]: any }) => data.PIN },
  PLAN_NUMBER: {
    label: 'Plan number',
    display: (data: { [key: string]: any }) => data.PLAN_NUMBER,
  },
  LEGAL_DESCRIPTION: {
    label: 'Legal description',
    display: (data: { [key: string]: any }) => data.LEGAL_DESCRIPTION,
  },
  OWNER_TYPE: { label: 'Owner type', display: (data: { [key: string]: any }) => data.OWNER_TYPE },
  MUNICIPALITY: {
    label: 'Municipality',
    display: (data: { [key: string]: any }) => data.MUNICIPALITY,
  },
  REGIONAL_DISTRICT: {
    label: 'Regional district',
    display: (data: { [key: string]: any }) => data.REGIONAL_DISTRICT,
  },
  FEATURE_AREA_SQM: {
    label: 'Area',
    display: (data: { [key: string]: any }) => (
      <>
        {data.FEATURE_AREA_SQM} m<sup>2</sup>
      </>
    ),
  },
  PARCEL_CLASS: {
    label: 'Parcel class',
    display: (data: { [key: string]: any }) => data.PARCEL_CLASS,
  },
  SURVEY_DESIGNATION_1: {
    label: 'Survey designation one',
    display: (data: { [key: string]: any }) => data.SURVEY_DESIGNATION_1,
  },
  SURVEY_DESIGNATION_2: {
    label: 'Survey designation two',
    display: (data: { [key: string]: any }) => data.SURVEY_DESIGNATION_2,
  },
  SURVEY_DESIGNATION_3: {
    label: 'Survey designation three',
    display: (data: { [key: string]: any }) => data.SURVEY_DESIGNATION_3,
  },
};

// Utility function that dynamically creates a ContentConfig for a feature.
// Transforms "THE_PARAM" to "The param"
export const getDynamicFeatureConfig: (feature: Feature) => ContentConfig = (feature: Feature) => {
  // For debug enable this
  /*
  config['Feature Type'] = {
    label: 'Feature Type',
    display: (data: { [key: string]: any }) => feature.id,
  };
  */
  const config: ContentConfig = {};
  Object.keys(feature.properties).forEach(key => {
    config[key] = {
      label: key.charAt(0).toUpperCase() + key.slice(1).toLowerCase().replaceAll('_', ' '),
      display: (data: { [key: string]: any }) => data[key],
    };
  });
  // This is not being mapped in the typed feature data
  if (exists(config['SE_ANNO_CAD_DATA'])) {
    delete config['SE_ANNO_CAD_DATA'];
  }
  return config;
};

export const municipalityLayerConfig: ContentConfig = {
  MUNICIPALITY_NAME: {
    label: 'Municipality Name',
    display: (data: { [key: string]: any }) =>
      `${data.ADMIN_AREA_GROUP_NAME} (${data.CHANGE_REQUESTED_ORG})`,
  },
};

export const pimsLayerConfig: ContentConfig = {
  PROPERTY_ID: {
    label: 'Hyperlink',
    display: (data: { [key: string]: any }) => (
      <a
        href={`/mapview/sidebar/property/${data.PROPERTY_ID}/details`}
        target="_blank"
        rel="noreferrer"
      >
        View PIMS property&nbsp;
        <FaExternalLinkAlt />
      </a>
    ),
  },
  PID: {
    label: 'PID',
    display: (data: { [key: string]: any }) => pidFormatter(data.PID_PADDED),
  },
  PIN: {
    label: 'PIN',
    display: (data: { [key: string]: any }) => data.PIN,
  },
  UNIQUE_ID: {
    label: 'Plan number',
    display: (data: { [key: string]: any }) => data.SURVEY_PLAN_NUMBER,
  },
  IS_RETIRED: {
    label: 'Retired ?',
    display: (data: { [key: string]: any }) => (data.IS_RETIRED ? 'Yes' : 'No'),
  },
  LAND_LEGAL_DESCRIPTION: {
    label: 'PIMS legal description',
    display: (data: { [key: string]: any }) => data.LAND_LEGAL_DESCRIPTION,
  },

  PIMS_ADDRESS: {
    label: 'Pims Address',
    display: (data: { [key: string]: any }) =>
      formatApiAddress({
        streetAddress1: data.STREET_ADDRESS_1,
        streetAddress2: data.STREET_ADDRESS_2,
        streetAddress3: data.STREET_ADDRESS_3,
        postal: data.POSTAL_CODE,
        province: { code: 'BC' },
        municipality: data.MUNICIPALITY_NAME,
      } as ApiGen_Concepts_Address), // force the type to avoid having to list unused fields.
  },
  LAND_AREA: {
    label: 'PIMS land area',
    display: (data: { [key: string]: any }) =>
      data.LAND_AREA + ' ' + getAreaUnit(data.PROPERTY_AREA_UNIT_TYPE_CODE.toLowerCase()),
  },
};

export const highwayLayerConfig: ContentConfig = {
  PROVINCIAL_PUBLIC_HIGHWAY_ID: {
    label: 'Id',
    display: (data: { [key: string]: any }) => data.PROVINCIAL_PUBLIC_HIGHWAY_ID,
  },
  GLOBALID: { label: 'Global Id', display: (data: { [key: string]: any }) => data.GLOBALID },
  UNIQUE_ID: { label: 'Unique Id', display: (data: { [key: string]: any }) => data.UNIQUE_ID },
  HYPERLINK: {
    label: 'Hyperlink to plan documents',
    display: (data: { [key: string]: any }) => (
      <a href={`${data.HYPERLINK}`} target="_blank" rel="noreferrer">
        {decodeURI(data.HYPERLINK)}
      </a>
    ),
  },
  PLAN_ANNOTATION: {
    label: 'Plan annotation',
    display: (data: { [key: string]: any }) => data.PLAN_ANNOTATION,
  },
  MOTI_PLAN: { label: 'MOTI plan', display: (data: { [key: string]: any }) => data.MOTI_PLAN },
  VETTING_STATUS: {
    label: 'Vetting Status',
    display: (data: { [key: string]: any }) => data.VETTING_STATUS,
  },
  SHAPE_TYPE: { label: 'Shape type', display: (data: { [key: string]: any }) => data.SHAPE_TYPE },
  PENDING_CLASSIFICATION: {
    label: 'Pending classification',
    display: (data: { [key: string]: any }) => data.PENDING_CLASSIFICATION,
  },
  MOTI_FILE: { label: 'MOTI file', display: (data: { [key: string]: any }) => data.MOTI_FILE },
  GAZETTE_PUBLISHED_DATE: {
    label: 'Gazette published date',
    display: (data: { [key: string]: any }) => data.GAZETTE_PUBLISHED_DATE,
  },
  GAZETTE_PUBLISHED_ON: {
    label: 'Gazette published on',
    display: (data: { [key: string]: any }) => data.GAZETTE_PUBLISHED_ON,
  },
  PROV_PUBLIC_HIGHWAY_TYPE: {
    label: 'Prov public highway type',
    display: (data: { [key: string]: any }) => data.PROV_PUBLIC_HIGHWAY_TYPE,
  },
  ORDER_IN_COUNCIL: {
    label: 'Order in council',
    display: (data: { [key: string]: any }) => data.ORDER_IN_COUNCIL,
  },
  SHORT_LEGAL_DESCRIPTION: {
    label: 'Short legal description',
    display: (data: { [key: string]: any }) => data.SHORT_LEGAL_DESCRIPTION,
  },
  SURVEY_PLAN: {
    label: 'Survey plan',
    display: (data: { [key: string]: any }) => data.SURVEY_PLAN,
  },
  PARENT_PARCEL: {
    label: 'Parent parcel',
    display: (data: { [key: string]: any }) => data.PARENT_PARCEL,
  },
  PARCELMAPBC_COMMENT: {
    label: 'Parcelmapbc comment',
    display: (data: { [key: string]: any }) => data.PARCELMAPBC_COMMENT,
  },
  CREATE_TIME: {
    label: 'Create time',
    display: (data: { [key: string]: any }) => data.CREATE_TIME,
  },
  UPDATE_TIME: {
    label: 'Update time',
    display: (data: { [key: string]: any }) => data.UPDATE_TIME,
  },
  CREATE_USER: {
    label: 'Create user',
    display: (data: { [key: string]: any }) => data.CREATE_USER,
  },
  UPDATE_USER: {
    label: 'Update user',
    display: (data: { [key: string]: any }) => data.UPDATE_USER,
  },
  LTSA_CREATE_TIME: {
    label: 'Ltsa create time',
    display: (data: { [key: string]: any }) => data.LTSA_CREATE_TIME,
  },
  LTSA_UPDATE_TIME: {
    label: 'Ltsa update time',
    display: (data: { [key: string]: any }) => data.LTSA_UPDATE_TIME,
  },
  LTSA_CREATE_USER: {
    label: 'Ltsa create user',
    display: (data: { [key: string]: any }) => data.LTSA_CREATE_USER,
  },
  LTSA_UPDATE_USER: {
    label: 'Ltsa update user',
    display: (data: { [key: string]: any }) => data.LTSA_UPDATE_USER,
  },
  RELATED_GAZETTES: {
    label: 'Related gazettes',
    display: (data: { [key: string]: any }) => data.RELATED_GAZETTES,
  },
  RELATIVE_PATH: {
    label: 'Relative path',
    display: (data: { [key: string]: any }) => data.RELATIVE_PATH,
  },
};
