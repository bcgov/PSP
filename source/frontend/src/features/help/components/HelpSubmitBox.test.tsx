import { act, render, screen, waitFor } from '@testing-library/react';

import { TicketTypes } from '../constants/HelpText';
import HelpSubmitBox from './HelpSubmitBox';

const renderHelpBox = (ticketType: TicketTypes) =>
  render(
    <HelpSubmitBox
      activeTicketType={ticketType}
      user="Test User"
      email="test@test.com"
      setActiveTicketType={jest.fn(x => x)}
      setMailto={jest.fn()}
      page="Landing Page"
    />,
  );

describe('Help Box tests', () => {
  it('renders properly', async () => {
    await act(async () => {
      const { asFragment } = renderHelpBox(TicketTypes.QUESTION);
      const fragment = await waitFor(() => asFragment());

      expect(fragment).toMatchSnapshot();
    });
  });

  it('contains question field and option when question selected', async () => {
    await act(async () => {
      renderHelpBox(TicketTypes.QUESTION);
    });

    const question = await waitFor(() => screen.getAllByText('Question'));
    expect(question).toHaveLength(2);
  });

  it('displays appropriate fields when switching to bug type', async () => {
    await act(async () => {
      renderHelpBox(TicketTypes.BUG);
    });
    const reproduce = await waitFor(() => screen.getByText('Steps to Reproduce'));
    const expected = await waitFor(() => screen.getByText('Expected Result'));
    const result = await waitFor(() => screen.getByText('Actual Result'));
    expect(reproduce).toBeInTheDocument();
    expect(expected).toBeInTheDocument();
    expect(result).toBeInTheDocument();
  });

  it('displays appropriate fields when switching to feature request ticket type', async () => {
    await act(async () => {
      renderHelpBox(TicketTypes.FEATURE_REQUEST);
    });
    const desc = await waitFor(() => screen.getByText('Description'));
    expect(desc).toBeInTheDocument();
  });
});
