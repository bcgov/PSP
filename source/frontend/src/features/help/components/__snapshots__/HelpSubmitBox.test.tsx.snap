// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help Box tests renders properly 1`] = `<DocumentFragment />`;
